sqladmin/forms.py

```diff
@@ -161,107 +199,169 @@ async def convert(
+    ) -> UnboundField:
+        kwargs.setdefault("render_kw", {})
+        kwargs["render_kw"]["class"] = "form-check-input"
+        return BooleanField(**kwargs)
 
     @converts("Date")
-    def conv_Date(self, field_args: Dict, **kwargs: Any) -> Field:
-        return DateField(**field_args)
+    def conv_Date(
+        self, model: type, prop: ColumnProperty, kwargs: Dict[str, Any]
+    ) -> UnboundField:
+        return DateField(**kwargs)
 
     @converts("DateTime")
-    def conv_DateTime(self, field_args: Dict, **kwargs: Any) -> Field:
-        return DateTimeField(**field_args)
+    def conv_DateTime(
+        self, model: type, prop: ColumnProperty, kwargs: Dict[str, Any]
+    ) -> UnboundField:
+        return DateTimeField(**kwargs)
 
     @converts("Enum")
-    def conv_Enum(self, column: Column, field_args: Dict, **kwargs: Any) -> Field:
-        available_choices = [(e, e) for e in column.type.enums]
+    def conv_Enum(
+        self, model: type, prop: ColumnProperty, kwargs: Dict[str, Any]
+    ) -> UnboundField:
+        available_choices = [(e, e) for e in prop.columns[0].type.enums]
         accepted_values = [choice[0] for choice in available_choices]
 
-        field_args["choices"] = available_choices
-        field_args["validators"].append(validators.AnyOf(accepted_values))
-        field_args["coerce"] = lambda v: v.name if isinstance(v, Enum) else str(v)
-        return SelectField(**field_args)
+        kwargs["choices"] = available_choices
+        kwargs.setdefault("validators", [])
+        kwargs["validators"].append(validators.AnyOf(accepted_values))
+        kwargs["coerce"] = lambda v: v.name if isinstance(v, Enum) else str(v)
+        return SelectField(**kwargs)
 
     @converts("Integer")  # includes BigInteger and SmallInteger
     def handle_integer_types(
-        self, column: Column, field_args: Dict, **kwargs: Any
-    ) -> Field:
-        return IntegerField(**field_args)
+        self, model: type, prop: ColumnProperty, kwargs: Dict[str, Any]
+    ) -> UnboundField:
+        return IntegerField(**kwargs)
 
     @converts("Numeric")  # includes DECIMAL, Float/FLOAT, REAL, and DOUBLE
     def handle_decimal_types(
-        self, column: Column, field_args: Dict, **kwargs: Any
-    ) -> Field:
+        self, model: type, prop: ColumnProperty, kwargs: Dict[str, Any]
+    ) -> UnboundField:
         # override default decimal places limit, use database defaults instead
-        field_args.setdefault("places", None)
-        return DecimalField(**field_args)
+        kwargs.setdefault("places", None)
+        return DecimalField(**kwargs)
 
     # @converts("dialects.mysql.types.YEAR", "dialects.mysql.base.YEAR")
-    # def conv_MSYear(self, field_args: Dict, **kwargs: Any) -> Field:
-    #     field_args["validators"].append(validators.NumberRange(min=1901, max=2155))
-    #     return StringField(**field_args)
+    # def conv_MSYear(
```

Comment:
I'd rather leave this out for now.

Reply:
I think we can avoid changing this. we can update it when adding it. Does that make sense?

Reply:
Sure